src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -293,9 +293,19 @@ private static bool TopLevelChanged(SyntaxTree oldTree, SourceText oldText, Synt
         /// </summary>
         public bool HasContentChanged(DocumentState oldState)
         {
-            return oldState._treeSource != this._treeSource
-                || oldState.sourceTextOpt != this.sourceTextOpt
-                || oldState.textAndVersionSource != this.textAndVersionSource;
+            if (oldState.sourceTextOpt != this.sourceTextOpt
+                || oldState.textAndVersionSource != this.textAndVersionSource)
+            {
+                return true;
+            }
+            else if (oldState._treeSource == this._treeSource)
```

Comment:
else if's not necessary.

Reply:
I am trying to save an async call for _treeSource.GetValue() if the _treeSource are same

Reply:
was it necessary to change this function here?  if so, why?

Reply:
we already has HasContentChanged (http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/DocumentState.cs,6a914fa8e24b0a45)

what's difference?